current/runtime/storage.md

```diff
@@ -56,13 +247,240 @@ decl_storage! {
 }
 ```
 
-## Verify First, Write Last
+### Genesis Configuration
 
-TODO
+Substrate's runtime storage APIs include capabilities to initialize storage items in the genesis
+block of your blockchain. The genesis storage configuration APIs expose a number of mechanisms for
+initializing storage, all of which have entry points in the `decl_storage` macro. These mechanisms
+all result in the creation of a `GenesisConfig` data type that implements
+[the `sp_runtime::BuildModuleGenesisStorage` trait](https://crates.parity.io/sp_runtime/trait.BuildModuleGenesisStorage.html)
+and will be added to the storage item's module (e.g.
+[`Struct pallet_balances::GenesisConfig`](https://crates.parity.io/pallet_balances/struct.GenesisConfig.html));
+storage items that are tagged for genesis configuration will have a corresponding attribute on this
+data type. In order to consume a module's genesis configuration capabilities, you must include the
+`Config` element when adding the module to your runtime with
+[the `construct_runtime` macro](https://crates.parity.io/frame_support/macro.construct_runtime.html).
+All the `GenesisConfig` types for the modules that inform a runtime will be aggregated into a single
+`GenesisConfig` type for that runtime, which implements
+[the `sp_runtime::BuildStorage` trait](https://crates.parity.io/sp_runtime/trait.BuildStorage.html)
+(e.g.
+[`Struct node_template_runtime::GenesisConfig`](https://crates.parity.io/node_template_runtime/struct.GenesisConfig.html));
+each attribute on this type corresponds to a `GenesisConfig` from one of the runtime's modules.
+Ultimately, the runtime's `GenesisConfig` is exposed by way of
+[the `sc_chain_spec::ChainSpec` trait](https://crates.parity.io/sc_chain_spec/trait.ChainSpec.html).
+For a complete and concrete example of using Substrate's genesis storage configuration capabilities,
+refer to the `decl_storage` macro in
+[the Society pallet](https://github.com/paritytech/substrate/blob/master/frame/society/src/lib.rs)
```

Comment:
should it be better to point to example pallet ?

Reply:
I don't believe that the example does any genesis configuration.